pkg/endpoint/policy.go

```diff
@@ -286,6 +286,9 @@ func (e *Endpoint) SetIdentity(owner Owner, id *policy.Identity) {
 	defer tree.Mutex.Unlock()
 	cache := owner.GetConsumableCache()
 
+	e.SecLabel = id
+	e.Consumable = cache.GetOrCreate(id.ID, id)
+
```

Comment:
Given that the containers/endpoints property needs to be udpated, we should always replace in the cache then.